scrapy_webarchive/wacz.py

```diff
@@ -86,23 +92,61 @@ def get_wacz_fname(self) -> str:
 
         return f"{self.collection_name}-{get_formatted_dt_string(format=TIMESTAMP_DT_FORMAT)}.wacz"
     
-    def write_datapackage(self, zip_file: zipfile.ZipFile):
-        package_dict = {
+    def write_datapackage(self, zip_file: zipfile.ZipFile) -> None:
+        """Main function to create and write the datapackage.json."""
+
+        package_dict = self.create_package_dict()
+
+        with zip_file.open("archive/" + self.warc_fname) as warc_file:
+            package_dict = self.update_package_metadata_from_warc(warc_file, package_dict)
+
+        package_dict["resources"] = self.collect_resources(zip_file)
+
+        zip_file.writestr(self.datapackage_fname, json.dumps(package_dict, indent=2))
+
+    def create_package_dict(self) -> Dict[str, Any]:
+        """Creates the initial package dictionary."""
+
+        dt_string = get_formatted_dt_string(format=WARC_DT_FORMAT)
+        return {
             "profile": "data-package",
             "title": self.collection_name,
-            "description": "This is the web archive generated by a Scrapy extension for the " \
-                f"{self.collection_name} spider. It is mainly for scraping purposes as it does not contain " \
-                "any js/css data. Though it can be replayed as bare HTML if the site does not depend on JS.",
-            "created": get_formatted_dt_string(format=WARC_DT_FORMAT),
-            "software": f"scrapy-webarchive v{scrapy_webarchive_version}, using Scrapy v{scrapy_version}",
-            "resources": [{}],
+            "description": "This is the web archive generated by a scrapy-webarchive extension for the " \
+                        f"{self.collection_name} spider. It is mainly for scraping purposes as it does not contain " \
+                        "any js/css data. Though it can be replayed as bare HTML if the site does not depend on JavaScript.",
+            "created": dt_string,
+            "modified": dt_string,
+            "wacz_version": WACZ_VERSION,
+            "software": f"Powered by scrapy-webarchive/{scrapy_webarchive_version}, using Scrapy/{scrapy_version}",
```

Comment:
Just to check, does other software have the _Powered by_ words?

Reply:
I could not find any specific examples or requirements for this. Perhaps we could just make it a comma separated list of software, like so:

`scrapy-webarchive/version, Scrapy/version`

Reply:
for the user-agent, this is space-separated iirc